adapters/Unity/UnityAdapter/GADMAdapterUnity.m

```diff
@@ -81,17 +80,26 @@ - (void)getInterstitial {
     [strongConnector adapter:self didFailAd:error];
     return;
   }
-  _isLoading = YES;
-  [[GADMAdapterUnitySingleton sharedInstance] requestInterstitialAdWithDelegate:self];
+
+  UnitySingletonCompletion completeBlock = ^(UnityAdsError *error, NSString *message) {
+    if(!error) {
+            [UnityAds addDelegate:self];
+            [UnityAds load:[self getPlacementID]];
+    } else {
+      id<GADMAdNetworkConnector> strongNetworkConnector = self->_networkConnector;
```

Comment:
Why use the `->` here and not `self._networkConnector`?

Reply:
I'm not a ObjC expert but is it correct way to convert weak to strong?

should it be something like:

if (self->_networkConnector) {
     __strong id<GADMAdNetworkConnector> strongNetworkConnector = self->_networkConnector;
}